File: tests/fixtures/xpath_injection/typescript/baseline_constant_xpath.ts
// Baseline: expression is a compile-time constant.  No taint reaches
// xpath.select so no XPATH_INJECTION finding fires.
import * as xpath from 'xpath';
import { DOMParser } from 'xmldom';

export function lookup(req: any, res: any): void {
    const doc = new DOMParser().parseFromString('<root/>');
    const nodes = xpath.select("//user[@role='admin']", doc);
    res.json(nodes);
}

File: tests/fixtures/xpath_injection/c/baseline_constant_xpath.c
/* Baseline: expression is a compile-time constant.  No taint reaches
 * xmlXPathEvalExpression so no XPATH_INJECTION finding fires. */
#include <libxml/xpath.h>

xmlXPathObjectPtr do_lookup(xmlXPathContextPtr ctx) {
    return xmlXPathEvalExpression((xmlChar *)"//user[@role='admin']", ctx);
}
